openthread/cmake/extensions.cmake

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Store the configuration of the compiled OpenThread libraries
+# and set source and destination paths.
+function(openthread_libs_configuration_write)
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND           ${GIT_EXECUTABLE} rev-parse HEAD
+      WORKING_DIRECTORY ${ZEPHYR_OPENTHREAD_MODULE_DIR}
+      RESULT_VARIABLE   git_return
+      OUTPUT_VARIABLE   openthread_git_hash
+      )
+
+    list(APPEND OPENTHREAD_SETTINGS "OpenThread_commit=${openthread_git_hash}\n\n")
+  endif()
+
+  # Store all OT related variables
+  get_cmake_property(_variableNames VARIABLES)
+  list (SORT _variableNames)
+  foreach (_variableName ${_variableNames})
+    if("${_variableName}" MATCHES "^CONFIG_OPENTHREAD_.*|^OT_.*")
+      list(APPEND OPENTHREAD_SETTINGS "${_variableName}=${${_variableName}}\n")
+    endif()
+  endforeach()
+
+  set(OPENTHREAD_CONFIG_FILE
+    "${CMAKE_BINARY_DIR}/openthread_lib_configuration.txt"
+    CACHE STRING "File containg OpenThread build configuration parameters.")
```

Comment:
The cache issue is still not addressed:
https://github.com/nrfconnect/sdk-nrf/pull/2655#discussion_r486278481

There is no reason to use a cached variable here, remember cached variable lives BETWEEN CMake invocations, and are not updated if already present in the cache.

Instead you can provide the conf file by the caller, and return dst dir.

```
function(openthread_libs_configuration_write CONFIG_FILE DST_DIR)
```
and then use `CONFIG_FILE` inside the function and return dst dir as:
```
set(${DST_DIR} "${ZEPHYR_NRFXLIB_MODULE_DIR}/openthread/${lib_path}v${CONFIG_OPENTHREAD_THREAD_VERSION}"
  PARENT_SCOPE
)

```

and then update ` openthread/CMakeLists.txt`:
```
set(OPENTHREAD_CONFIG_FILE
    "${CMAKE_BINARY_DIR}/openthread_lib_configuration.txt"
)
openthread_libs_configuration_write(${OPENTHREAD_CONFIG_FILE} OPENTHREAD_DST_DIR)
```
inside the function.

You can also decide to return both variables from the function, or return just the OpenThread version from the function (instead of the), and then let the caller construct the dst dir using the returened version value.
Returning just the OpenThread version remove the need for the function to know about directory structures, which generally is nice as it makes the function more generic and reusable for other purposes.

Reply:
actually, the version seems to be a Kconfig setting, so no reason it should be inside the function:
`v${CONFIG_OPENTHREAD_THREAD_VERSION}` <-- Assume this is Kconfig value, correct ?

Reply:
Correct. No cached variables, passing only the `CONFIG_FILE` now.